### DISCUSSION TITLE
Can smack generate boogie code which saves the metadata in llvm ir?

### DISCUSSION THREAD

**Discussion by Luweicai** [smackers/smack#786] [Q&A] [answered]
How to save the metadata information in llvm ir during the transformation from llvm ir to boogie code?

**Comment by zvonimir** [comment, marked as answer]
Thanks for your question. Could you please clarify what you mean by this and/or maybe give an example?

In short, SMACK already uses certain kinds of LLVM IR metadata when generating Boogie code. For example, we use LLVM IT debug info to add debug (source code lines, variable names, etc.) info into the generated Boogie code. So we can certainly generate Boogie code based on the LLVM IR metadata.

Is that your question?
